tests/csm/rest/test_iam_users.py

```diff
@@ -386,7 +409,1460 @@ def test_35607(self):
+        user_id, display_name = self.get_IAM_user_payload()
+        payload = {"uid": user_id, "display_name": display_name, "generate_key": False}
+        self.log.info("payload :  %s", payload)
+        res = self.csm_obj.create_iam_user_rgw(payload)
+        assert res.status_code == HTTPStatus.CREATED, "Status code check failed for user creation"
+
+        self.log.info("Perform PUT request to add invalid capability")
+        payload = {"user_caps":"random=*;buckets=*"}
+        response = self.csm_obj.add_user_caps_rgw(user_id, payload)
+        assert response.status_code == HTTPStatus.BAD_REQUEST, "Status code check failed for add capability."
+        
+        self.log.info("Perform PUT request to add invalid  caps value")
+        payload = {"user_caps":"users=random;buckets=*"}
+        response = self.csm_obj.add_user_caps_rgw(user_id, payload)
+        assert response.status_code == HTTPStatus.OK, "Status code check failed for add capability."
+
+        self.log.info("Perform PUT request to add invalid  format caps")
+        payload = {"user_caps":"users=read,buckets=*"}
+        response = self.csm_obj.add_user_caps_rgw(user_id, payload)
+        assert response.status_code == HTTPStatus.OK, "Status code check failed for add capability."
+        
+        self.log.info("[END] Testing that IAM user can not add invalid capabilities.")
+    
+    @pytest.mark.csmrest
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-39127')
+    def test_39127(self):
+        """
+            Test that IAM user can not remove invalid capabilities.
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("[START] Testing that IAM user can not remove invalid capabilities.")
+        self.log.info("Create IAM user1.")
+        user_id, display_name = self.get_IAM_user_payload()
+        payload = {"uid": user_id, "display_name": display_name, "generate_key": False}
+        self.log.info("payload :  %s", payload)
+        res = self.csm_obj.create_iam_user_rgw(payload)
+        assert res.status_code == HTTPStatus.CREATED, "Status code check failed for user creation"
+
+        self.log.info("Perform PUT request to add capability for above user1 with admin rights")
+        payload = {"user_caps":"usage=read,write;user=write"}
+        response = self.csm_obj.add_user_caps_rgw(user_id, payload)
+        assert response.status_code == HTTPStatus.OK, "Status code check failed for add capability."
+        
+        self.log.info("Perform PUT request to remove invalid capability")
+        payload = {"user_caps":"random=*;buckets=*"}
+        response = self.csm_obj.remove_user_caps_rgw(user_id, payload)
+        assert response.status_code == HTTPStatus.BAD_REQUEST, "Status code check failed for add capability."
+        
+        self.log.info("Perform PUT request to remove invalid  caps value")
```

Comment:
Is this correct log for below statement?

Reply:
Resolved.